src/test/java/ut/com/lmig/forge/stash/ssh/keys/EnterpriseSshKeyManagerImplTest.java

```diff
@@ -0,0 +1,211 @@
+        when(userService.isUserInGroup(blessedUser, AUTHED_GROUP)).thenReturn(true);
+    }
+
+    @Test
+    public void whenExpireTaskIsCalledValidKeysAreIgnored() {
+        SshKeyEntity validKey = ao.get(SshKeyEntity.class, EnterpriseSshKeyRepositoryTestData.validKey.getID());
+        assertThat(validKey, notNullValue());
+        assertThat(validKey.getKeyId(), is(VALID_STASH_KEY_ID));
+
+        ourKeyService.replaceExpiredKeysAndNotifyUsers();
+
+        // key survived?
+        validKey = ao.get(SshKeyEntity.class, EnterpriseSshKeyRepositoryTestData.validKey.getID());
+        assertThat(validKey, notNullValue());
+        // stash's ssh service was not invoked
+        verify(stashKeyService, times(0)).remove(VALID_STASH_KEY_ID);
+    }
+
+    @Test
+    @NonTransactional
+    public void whenExpireTaskIsCalledExpiredKeysAreDestroyed() {
+        SshKeyEntity validKey = ao.get(SshKeyEntity.class, EnterpriseSshKeyRepositoryTestData.expiredKey.getID());
+        assertThat(validKey, notNullValue());
+        assertThat(validKey.getKeyId(), is(EXPIRED_STASH_KEY_ID));
+
+        ourKeyService.replaceExpiredKeysAndNotifyUsers();
+
+        // key was purged?
+        validKey = ao.get(SshKeyEntity.class, EnterpriseSshKeyRepositoryTestData.expiredKey.getID());
+        assertThat(validKey, nullValue());
+        // stash ssh remve was called with ecxpired ssh key id
+
+        // stash's ssh service was not invoked
+        verify(stashKeyService).remove(EXPIRED_STASH_KEY_ID);
+    }
+
+    @Test
+    public void whenKeyIsExpiredTheAppropriateUserIsNotified() {
+        ourKeyService.replaceExpiredKeysAndNotifyUsers();
+
+        // stash's ssh service was not invoked
+        verify(notificationService).notifyUserOfExpiredKey(EXPIRED_USER_ID);
+    }
+    
+    
+    @Test
+    public void userInBlessedGroupMayByPassDirectService(){
+        SshKey newKey = new SimpleSshKey(1, "a comment", UNAPPROVED_PUBLIC_KEY_ONE,blessedUser );
+        boolean isAllowed = ourKeyService.isKeyValidForUser(newKey, blessedUser);
+        
+        assertThat(isAllowed,is(true));
+    }
+    
+    
+    @Test
+    public void userNotInBlessedGroupAreNotAllowedDirect(){
+        
+        SshKey newKey = new SimpleSshKey(1, "a comment", UNAPPROVED_PUBLIC_KEY_ONE ,unblessedUser );
+        boolean isAllowed = ourKeyService.isKeyValidForUser(newKey, unblessedUser);
+        
+        assertThat(isAllowed,is(false));        
+    }
+
+    @Test
+    public void userNotInBlessedGroupButUsedWrapperServiceAreAllowed(){
+        //when(keyRepo.isValidKeyForUser(unblessedUser, PUBLIC_KEY_ONE)).thenReturn(true);
+        SshKey newKey = new SimpleSshKey(1, "a comment", APPROVED_PUBLIC_KEY_ONE,unblessedUser );
+        boolean isAllowed = ourKeyService.isKeyValidForUser(newKey, unblessedUser);
+        
+        assertThat(isAllowed,is(true));        
+    }
+    
+ 
+    public static class EnterpriseSshKeyRepositoryTestData implements DatabaseUpdater {
```

Comment:
I think this test class can be cleaned up to push the keyRepo responsibilities to the KeyRepotests, and bring the level of testing up/away from persistence details.  Some methods in keyRepoTest may already cover these, or can be moved over to that class.